permuta/permutation_sets.py

```diff
@@ -17,6 +17,37 @@ def __repr__(self):
     def __len__(self):
         return 1
 
+class AvoidanceClass(object):
+    def __new__(cls, n, avoiding=None):
+        if isinstance(avoiding, Permutation):
```

Comment:
What if avoiding is not a Permutation? It would be neat if this code could handle arrays of Permutations as well (i.e. avoiders of a set of permutations). So, if the array consists of a single element, then you perform the same check as below, otherwise you defer to the generic class.